temporian/core/operators/drop.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2021 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Drop operator class and public API function definitions."""
+
+from typing import List
+from temporian.core import operator_lib
+from temporian.core.compilation import compile
+from temporian.core.data.node import (
+    EventSetNode,
+    create_node_new_features_new_sampling,
+)
+from temporian.core.operators.base import Operator
+from temporian.core.typing import EventSetOrNode
+from temporian.proto import core_pb2 as pb
+from temporian.utils.typecheck import typecheck
+from temporian.core.operators.select import select
+
+
+@typecheck
+@compile
+def drop(
+    input: EventSetOrNode, feature_names: str | List[str]
```

Comment:
The `|` syntax for type union was only introduced in 3.10, and temporian supports 3.8 and up (which is why the [tests GH action is failing](https://github.com/google/temporian/actions/runs/6895848158/job/18760722564?pr=313) on py3.8) - we need to use `Union`